### PR TITLE
feat: allow users to check cpu arch 

### PR DESCRIPTION
**Pull request by 5hubh49**
## Description, Motivation and Context
Story details: https://app.shortcut.com/replicated/story/111542/add-support-for-cpu-architecture-to-cpu-host-collector-analyser-1584

Demo:

https://github.com/user-attachments/assets/17987849-216a-4ef5-b7c4-e1044afe506e

## Checklist

- [x] New and existing tests pass locally with introduced changes.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) are informative and highlight any breaking changes
- [ ] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

This feature helps to find which CPU architecture the machine is running on.

### CODE REVIEW

File: pkg/analyze/host_cpu.go [hunk 1]
```diff
@@ -164,14 +164,19 @@ func doCompareHostCPUFlags(expected string, flags []string) (res bool, err error
...
+func compareHostCPUConditionalToActual(conditional string, logicalCount int, physicalCount int, flags []string, machineArch string) (res bool, err error) {
 	compareLogical := false
 	comparePhysical := false
 	compareUnspecified := false
+	compareMachineArch := false
 
 	comparator := ""
 	desired := ""
 
+	/* When the conditional is in the format of "logical <comparator> <desired>"
```
Comment (banjoh) [inline pkg/analyze/host_cpu.go:176]:
```suggestion
	/* when conditional format is "<field_identifier> <comparator> <desired>"
```

File: pkg/collect/host_cpu_test.go [hunk 2]
```diff
@@ -29,4 +29,5 @@ func TestCollectHostCPU_Collect(t *testing.T) {
 	assert.Contains(t, m, "logicalCount")
 	assert.Contains(t, m, "physicalCount")
 	assert.Contains(t, m, "flags")
+	//assert.Contains(t, m, "machineArch")
```
Comment (banjoh) [inline pkg/collect/host_cpu_test.go]: Should this be uncommented?

**Review by banjoh — COMMENTED** [review]
Changes look good. A unit test would be good to add

File: pkg/collect/host_cpu_test.go (hunk 2 above)
Comment (5hubh49) [reply pkg/collect/host_cpu_test.go]: done

**Review by banjoh — APPROVED** [review]
LGTM

**Pull request merged**